chia/wallet/nft_wallet/uncurry_nft.py

```diff
@@ -94,18 +94,21 @@ def uncurry(cls: Type[_T_UncurriedNFT], puzzle: Program) -> _T_UncurriedNFT:
         """
         mod, curried_args = puzzle.uncurry()
         if mod != SINGLETON_TOP_LAYER_MOD:
-            raise ValueError(f"Cannot uncurry NFT puzzle, failed on singleton top layer: Mod {mod}")
```

Comment:
generating this f-string is the expensive part. this patch still allows it to be printed, but only when debug logging is enabled.